cluster-controller/src/main/java/io/strimzi/controller/cluster/OpenShiftUtils.java

```diff
@@ -0,0 +1,162 @@
+package io.strimzi.controller.cluster;
+
+import io.fabric8.kubernetes.api.builder.BaseFluent;
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.DoneableConfigMap;
+import io.fabric8.kubernetes.api.model.DoneablePersistentVolumeClaim;
+import io.fabric8.kubernetes.api.model.DoneablePod;
+import io.fabric8.kubernetes.api.model.DoneableService;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.KubernetesResource;
+import io.fabric8.kubernetes.api.model.PersistentVolumeClaim;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+import io.fabric8.kubernetes.api.model.extensions.DoneableDeployment;
+import io.fabric8.kubernetes.api.model.extensions.DoneableStatefulSet;
+import io.fabric8.kubernetes.api.model.extensions.StatefulSet;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.Watch;
+import io.fabric8.kubernetes.client.Watcher;
+import io.fabric8.kubernetes.client.dsl.Patchable;
+import io.fabric8.kubernetes.client.dsl.PodResource;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.fabric8.kubernetes.client.dsl.RollableScalableResource;
+import io.fabric8.kubernetes.client.dsl.ScalableResource;
+import io.fabric8.openshift.api.model.BuildConfig;
+import io.fabric8.openshift.api.model.ImageStream;
+import io.fabric8.openshift.client.OpenShiftClient;
+import io.fabric8.openshift.client.dsl.BuildConfigResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Utility class for handling OpenShift resources
+ */
+public class OpenShiftUtils {
+    private static final Logger log = LoggerFactory.getLogger(OpenShiftUtils.class.getName());
+
+    private final OpenShiftClient client;
+
+    /**
+     * Constructor
+     *
+     * @param client    Fabric8 OpenShift client
+     */
+    public OpenShiftUtils(OpenShiftClient client) {
+        this.client = client;
+    }
+
+    /**
+     * Returns the Fabric8 OpenShift client for direct use
+     *
+     * @return
+     */
+    public OpenShiftClient getOpenShiftClient() {
+        return client;
+    }
+
+    /**
+     * Creates OpenShift resource
+     *
+     * @param res   Resource which should be created
+     */
+    public void create(HasMetadata res) {
+        log.info("Creating {} {}", res.getClass().getSimpleName(), res.getMetadata().getName());
+
+        if (res instanceof BuildConfig) {
+            client.buildConfigs().createOrReplace((BuildConfig)res);
+        } else if (res instanceof ImageStream) {
+            client.imageStreams().createOrReplace((ImageStream)res);
+        } else {
+            throw new RuntimeException("Unsupported type " + res.getClass().getSimpleName());
+        }
+    }
```

Comment:
Yeah, I didn't liked how the `K8SUtils` has million of methods for the same task. So I tried it differently in OpenShiftUtils ... I leave it up to further discussion whether we want to change it this way in K8SUtils as well or maybe change OpenShiftUtils the same way as K8SUtils is ... WDYT? In any case, this would be separate PR.

Unfortunately I didn't found a way how to it without the _if ... else if ... else if ...else if ... else_ construct.

Reply:
I'm absolutely fine. It was just a question for better understanding the two different approaches. We'll discuss about that and in any case a new PR is needed.